src/main/java/com/nutomic/syncthingandroid/preferences/WifiSsidPreference.java

```diff
@@ -0,0 +1,107 @@
+package com.nutomic.syncthingandroid.preferences;
+
+import android.content.Context;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Bundle;
+import android.preference.MultiSelectListPreference;
+import android.util.AttributeSet;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import com.nutomic.syncthingandroid.R;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * MultiSelectListPreference which allows the user to select on which WiFi networks (based on SSID)
+ * syncing should be allowed.
+ *
+ * Setting can be "All networks" (none selected), or selecting individual networks.
+ *
+ * Due to restrictions in Android, it is possible/likely, that the list of saved WiFi networks
+ * cannot be retrieved if the WiFi is turned off. In this case, an explanation is shown.
+ *
+ * The preference is stored as Set&lt;String&gt; where an empty set represents
+ * "all networks allowed".
+ *
+ * SSIDs are formatted according to the naming convention of WifiManager, i.e. they have the
+ * surrounding double-quotes (") for UTF-8 names, or they are hex strings (if not quoted).
+ */
+public class WifiSsidPreference extends MultiSelectListPreference {
+
+    public WifiSsidPreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        setDialogLayoutResource(R.layout.preference_wifi_ssids);
+    }
+
+    @Override
+    protected void showDialog(Bundle state) {
+        WifiConfiguration[] networks = loadConfiguredNetworksSorted();
+        if (networks != null) {
+            Set<String> selected = getSharedPreferences().getStringSet(getKey(), null);
+            // from JavaDoc: Note that you must not modify the set instance returned by this call.
+            // therefore required to make a defensive copy of the elements
+            selected = selected != null ? new HashSet<>(selected) : new HashSet<String>();
```

Comment:
How about putting `new HashSet<>(getSharedPreferences().getStringSet(getKey(), new HashSet<String>());`?

And no need to quote the docs, I believe you now :D